src/Core/src/Platform/iOS/KeyboardAutoManager.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections.Generic;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class KeyboardAutoManager
+{
+	// you can provide a topView argument or it will use the most top Superview
+	internal static void GoToNextResponderOrResign(UIView view, UIView? topView = null)
+	{
+		if (!view.CheckIfEligible())
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var textFields = GetDeepResponderViews(topView ?? view.FindTopView());
+
+		// get the index of the current textField and go to the next one
+		var currentIndex = textFields.FindIndex(v => v == view);
+		var nextIndex = currentIndex < textFields.Count - 1 ? currentIndex + 1 : -1;
+
+		if (nextIndex != -1)
+			textFields[nextIndex].BecomeFirstResponder();
+		else
+			view.ResignFirstResponder();
+	}
+
+	static bool CheckIfEligible(this UIView view)
+	{
+		if (view is UITextField field && field.ReturnKeyType == UIReturnKeyType.Next)
+			return true;
+		else if (view is UITextView)
+			return true;
+
+		return false;
+	}
+
+	static UIView FindTopView (this UIView view)
+	{
+		var curView = view;
+
+		while (curView.Superview is not null)
+		{
+			curView = curView.Superview;
+		}
+
+		return curView;
+	}
+
+	// Find all of the eligible UITextFields and UITextViews inside this view
+	internal static List<UIView> GetDeepResponderViews(UIView view)
```

Comment:
One thing to note for folks reviewing this PR.  The implementation in XF was similar-ish. Whenever you hit "Next" it would build the entire TabIndex tree from every element on the screen. So, this hierarchical search isn't super different than what we were doing in XF.

Reply:
For .NET8 we are but this is a big complaint area right now for .NET7 and people migrating for Forms 

So, we will most likely backport a portion of this for .NET7 and then the API for .NET8 is to make it configurable